bliss/surveys/sdss.py

```diff
@@ -345,12 +378,157 @@ def restrict_by_ra_dec(self, ra_lim, dec_lim):
             d[key] = val[:, keep]
 
         return PhotoFullCatalog(
-            plocs[0, :, 0].max() - plocs[0, :, 0].min(),  # new height
-            plocs[0, :, 1].max() - plocs[0, :, 1].min(),  # new width
+            int(plocs[0, :, 0].max() - plocs[0, :, 0].min()),  # new height
+            int(plocs[0, :, 1].max() - plocs[0, :, 1].min()),  # new width
             d,
         )
 
 
+class SDSSPointSpreadFunction(ImagePointSpreadFunction):
+    def __init__(self, survey_data_dir, image_ids, bands, psf_config: PSFConfig):
+        super().__init__(bands, **psf_config)
+
+        self.psf_galsim = {}
+        self.psf_params = {}
+        for run, camcol, field in image_ids:
+            # load raw params from file
+            field_dir = f"{survey_data_dir}/{run}/{camcol}/{field}"
+            filename = f"{field_dir}/psField-{run:06}-{camcol}-{field:04}.fits"
+            if not Path(filename).exists():
+                SDSSDownloader(run, camcol, field, download_dir=survey_data_dir).download_psfield()
+            psf_params = self._get_fit_file_psf_params(filename, bands)
+
+            # load psf image from params
+            self.psf_params[(run, camcol, field)] = psf_params
+            self.psf_galsim[(run, camcol, field)] = self._get_psf(psf_params)
+
+    @staticmethod
+    def _get_fit_file_psf_params(psf_fit_file: str, bands: Tuple[int, ...]):
+        """Load psf parameters from fits file.
+
+        See https://data.sdss.org/datamodel/files/PHOTO_REDUX/RERUN/RUN/objcs/CAMCOL/psField.html
+        for details on the parameters.
+
+        Args:
+            psf_fit_file (str): file to load from
+            bands (Tuple[int, ...]): SDSS bands to load
+
+        Returns:
+            psf_params: tensor of parameters for each band
+        """
+        msg = (
+            f"{psf_fit_file} does not exist. "
+            + "Make sure data files are available for fields specified in config."
+        )
+        assert Path(psf_fit_file).exists(), msg
+        # HDU 6 contains the PSF header (after primary and eigenimages)
+        data = fits.open(psf_fit_file, ignore_missing_end=True).pop(6).data
+        psf_params = torch.zeros(len(bands), 6)
+        for i, band in enumerate(bands):
+            sigma1 = data["psf_sigma1"][0][band] ** 2
+            sigma2 = data["psf_sigma2"][0][band] ** 2
+            sigmap = data["psf_sigmap"][0][band] ** 2
+            beta = data["psf_beta"][0][band]
+            b = data["psf_b"][0][band]
+            p0 = data["psf_p0"][0][band]
+
+            psf_params[i] = torch.tensor([sigma1, sigma2, sigmap, beta, b, p0])
+
+        return psf_params
+
+
+class SDSSPrior(ImagePrior):
```

Comment:
It's kinda confusing to think of each survey as having it's own prior. (The prior is part of a statistical model, not a survey.)

Reply:
Fortunately, I believe @sawanp813 is replacing most/all the code in this code with a Gaussian mixture model color prior, which will be fitted offline. So perhaps we'll be able to drop this class soon.